pkg/appmanager/as3Manager_test.go

```diff
@@ -19,30 +19,83 @@ import (
 	"net/http"
 	"net/http/httptest"
 
+	"github.com/F5Networks/k8s-bigip-ctlr/pkg/test"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+
+	"k8s.io/client-go/kubernetes/fake"
 )
 
-var _ = Describe("Create HTTP REST mock client and test POST call", func() {
-
-	It("AS3 declaratyion POST call test with 200 OK response", func() {
-		route := "/mgmt/shared/appsvcs/declare"
-		method := "POST"
-		var template as3Declaration = `{"class":"AS3","action":"deploy","persist":true,}`
-		server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-			// Test request parameters
-			Expect(req.URL.String()).To(BeEquivalentTo("/mgmt/shared/appsvcs/declare"))
-			// Send response to be tested
-			_, err := rw.Write([]byte("OK"))
-			Expect(err).To(BeNil(), "Response writer should be written.")
-		}))
-		// Close the server when test finishes
-		defer server.Close()
-		// Use Client & URL from our local test server
-		api := As3RestClient{server.Client(), server.URL}
-		body, status := api.restCallToBigIP(method, route, template)
-		Expect(status).To(BeTrue())
-		Expect(body).To(BeEquivalentTo("OK"), "Response should be captured.")
+var _ = Describe("As3Manager Tests", func() {
+	Describe("Validating AS3 ConfigMap with AS3Manager", func() {
+		var mockMgr *mockAppManager
+		var mw *test.MockWriter
+		BeforeEach(func() {
+			RegisterBigIPSchemaTypes()
+
+			mw = &test.MockWriter{
+				FailStyle: test.Success,
+				Sections:  make(map[string]interface{}),
+			}
+			fakeClient := fake.NewSimpleClientset()
+			Expect(fakeClient).ToNot(BeNil())
+
+			mockMgr = newMockAppManager(&Params{
+				KubeClient:       fakeClient,
+				ConfigWriter:     mw,
+				restClient:       test.CreateFakeHTTPClient(),
+				RouteClientV1:    test.CreateFakeHTTPClient(),
+				IsNodePort:       true,
+				broadcasterFunc:  NewFakeEventBroadcaster,
+				ManageConfigMaps: true,
+			})
+		})
+		AfterEach(func() {
+			mockMgr.shutdown()
+		})
+
```

Comment:
Could you add a test case that passes a JSON file with errors and expects this error? This should make the coverage 100%
https://github.com/F5Networks/k8s-bigip-ctlr/blob/794b182fe2756db550fda700fd76579b2679a177/pkg/appmanager/as3Manager.go#L63